concert/app/src/functions/on_event_stage_changed/index.py

```diff
@@ -88,3 +104,107 @@ def get_song_list_update_transaction_item(message_data):
     }
 
     return {"Update": song_list_update_args}
+
+
+def get_song_list_get_transaction_item():
+    serializer = boto3.dynamodb.types.TypeSerializer()
+    song_list_get_args = {
+        "Key": {ATTR_RECORD_ID: serializer.serialize(RECORD_ID_SONG_LIST)},
+        "TableName": DYNAMODB_TABLE_NAME,
+    }
+
+    return {"Get": song_list_get_args}
+
+
+def get_aggregate_data_get_transaction_item():
+    serializer = boto3.dynamodb.types.TypeSerializer()
+    aggregate_data_get_args = {
+        "Key": {ATTR_RECORD_ID: serializer.serialize(RECORD_ID_AGGREGATE)},
+        "TableName": DYNAMODB_TABLE_NAME,
+    }
+
+    return {"Get": aggregate_data_get_args}
+
+
+def get_song_list_and_aggregate_data():
+    song_list_get_transaction_item = get_song_list_get_transaction_item()
+    aggregate_data_get_transaction_item = get_aggregate_data_get_transaction_item()
+
+    transaction_items = [
+        song_list_get_transaction_item,
+        aggregate_data_get_transaction_item,
+    ]
+
+    responses = dynamodb.transact_get_items(TransactItems=transaction_items)[
+        "Responses"
+    ]
+
+    deserializer = boto3.dynamodb.types.TypeDeserializer()
+    songs = {k: deserializer.deserialize(v) for k, v in responses[0]["Item"].items()}
+    aggregate_data = {
+        k: deserializer.deserialize(v) for k, v in responses[1]["Item"].items()
+    }
+
+    return songs, aggregate_data
+
+
+def get_songs_with_aggregate_choices(song_list, aggregate_data):
+    songs_with_aggregate_choices = []
+    for song_metadata in song_list[ATTR_SONG_LIST_SONGS]:
+        display_name = song_metadata[ATTR_SONG_LIST_DISPLAY_NAME]
+        start_time = song_metadata[ATTR_SONG_LIST_START_TIME]
+        end_time = song_metadata[ATTR_SONG_LIST_END_TIME]
+
+        choice_type = None
+
+        for timestamp in aggregate_data[ATTR_CHOICE_VALUE_COLOR].keys():
+            if start_time <= timestamp < end_time:
+                choice_type = ATTR_CHOICE_VALUE_COLOR
+                break
+        if not choice_type:
+            choice_type = ATTR_CHOICE_VALUE_CHILLS
+
+        choices = []
+        for timestamp, aggregate_choice_data in aggregate_data[choice_type].items():
+            if start_time <= timestamp < end_time:
+                choices_at_timestamp = {ATTR_CHOICE_TIMESTAMP: timestamp}
+                if choice_type == ATTR_CHOICE_VALUE_COLOR:
+                    choices_at_timestamp.update(
+                        {
+                            k[4:]: v
```

Comment:
If the aggregator didn't save with the sum_ prefix, lines 170-190 could be replaced with:
```py
choices.append({ ATTR_CHOICE_TIMESTAMP: timestamp, **aggregate_choice_data })
```

Reply:
This isn't quite true, as we'd still need to remove the `count` attribute from the color rows (either here or in the aggregate writer). IMO, that update would introduce as much messiness as we'd clean up here, so I'd prefer to leave this as is. (Though I'm not hugely opposed if you feel strongly.)